### PR TITLE
refactor: refactor mqtt _connect() method

### LINKED ISSUE #397: Refactor mqtt `_connect()` method
#### Reason/Context
the [_connect()](https://github.com/asyncapi/glee/blob/master/src/adapters/mqtt/index.ts#L33) method in `mqtt` adapter currently has a cognitive complexity of 29. It's really hard to understand what is going on. plus, SONAR always complains about it in PRs.

#### Description
refactor the `_connect()` method by grouping it's logic in separate functions for the purpose of readability and maintainability.

### PR DESCRIPTION
**Pull request by oviecodes**
**Description**
Refactor the mqtt adapter _connect() method

**Related issue(s)**
fixes #397

### CODE REVIEW

**Review by Souvikns — APPROVED** [review]
Good job @oviecodes! Looks neat 👍🏼

**Review by KhudaDad414 — CHANGES_REQUESTED** [review]
Left a few comments.
Regarding the package-lock.json Please regenerate it with a newer version of NPM that uses lockfile version 2.

File: src/adapters/mqtt/index.ts
Comment (oviecodes) [reply src/adapters/mqtt/index.ts:167]: I was thinking for future purposes, in case there was a need to pass other variables. I could change it if it's not necessary.

**Comment by oviecodes** [comment]
@KhudaDad414 I've made the requested changes

**Review by KhudaDad414 — APPROVED** [review]
LGTM! let's wait for @fmvilas approval.

**Comment by oviecodes** [comment]
@KhudaDad414 , the package-lock.json in the glee repo is using lockfile version 2.

**Comment by KhudaDad414** [comment]
@oviecodes the tests are failing 🤔

**Comment by oviecodes** [comment]
> @oviecodes the tests are failing 🤔

I'm trying to figure out why, I'm looking at it right now @KhudaDad414

**Comment by oviecodes** [comment]
I fixed it @KhudaDad414 , it was a problem with package-lock.json file. The test should pass now

**Comment by KhudaDad414** [comment]
hey @oviecodes Please resolve the conflicts so we can merge this. :)

**Comment by oviecodes** [comment]
@KhudaDad414 done

**Review by KhudaDad414 — APPROVED** [review]
LGTM!

**Pull request merged**
